queries.sql

```diff
@@ -7,4 +7,62 @@ SELECT name, date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
 SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = TRUE;
 SELECT * FROM animals WHERE name != 'Gabumon';
-SELECT * FROM animals WHERE weight_kg = 10.4 AND weight_kg = 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg = 10.4 AND weight_kg = 17.3;
+
+UPDATE animals SET species = 'unspecified';
+
+/*  Unspecified Rollback */
+
+-- To start the transaction
+BEGIN;
+
+-- Updating the table
+UPDATE animals SET species = 'unspecified';
+
+-- Checking the Records
+SELECT * FROM animals;
+
+-- Rollback to Earlier state
+ROLLBACK;
+
+-- Verify the changes have been reverted
+SELECT * FROM animals; 
+
+
+/* Setting Species */
+BEGIN; -- Starting Transaction
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon'; 
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
+
+/* Delete all records then Rollback */
+BEGIN;
+DELETE FROM animals; --Delete all records
+SELECT * FROM  animals; -- Verify if records have been deleted
+ROLLBACK; -- RollBack
+SELECT * FROM  animals; -- Verify if records have been restored
```

Comment:
- [OPTIONAL] Rather than selecting all fields in the table using `*`,  select the `species` column that was modified.

Reply:
In this case, it deleted all columns, not just the `species` column.